### ISSUE TITLE
[BUG] Allow repeated nested group labels

### ISSUE THREAD

**Issue by wkeese** [IBMa/equal-access#461]
## What tool(s) are you using?

- [ ] `accessibility-checker-engine`
- [X] `accessibility-checker`
- [ ] `karma-accessibility-checker`
- [ ] `rule-server`
- [ ] `boilerplates`
- [ ] `other`

## Detailed description

> Node version:

N/A

> Platform:

Chrome 90.

> Describe in detail the issue you're having.

In the sample below, there are two groups with unique labels.  Each group contains two sub-groups.  The sub-groups have unique labels _within their parent group_.

But EAAC throws an error because every group is considered in a global context.  I don't think it should throw an error.

The same reasoning applies if the parent container is a `role=form` or `role=search`.  You should be able to have two fieldsets (or `role=group`) with the same name if they are in different forms.

```html
<div role="group" aria-labelledby="groupLabel1">
	<strong id="groupLabel1">group 1 label</strong>
	<span id="label1">From</span>
	<div role="group" aria-labelledby="label1">...</div>
	<span id="label2">To</span>
	<div role="group" aria-labelledby="label2">...</div>
</div>

<div role="group" aria-labelledby="groupLabel2">
	<strong id="groupLabel2">group 2 label</strong>
	<span id="label3">From</span>
	<div role="group" aria-labelledby="label3">...</div>
	<span id="label4">To</span>
	<div role="group" aria-labelledby="label4">...</div>
</div>
```

## Additional information

Here's the full test case:

```html
<html lang="en">
<head>
	<title>Test for spurious EAAC warning</title>
</head>
<body>
<main>
	<div role="group" aria-labelledby="groupLabel1">
		<strong id="groupLabel1">group 1 label</strong>
		<div>
			<span id="label1">From</span>
			<div role="group" aria-labelledby="label1">
				<input aria-label="month">/<input aria-label="day">/<input aria-label="year">
			</div>
			<span id="label2">To</span>
			<div role="group" aria-labelledby="label2">
				<input aria-label="month">/<input aria-label="day">/<input aria-label="year">
			</div>
		</div>
	</div>

	<div role="group" aria-labelledby="groupLabel2">
		<strong id="groupLabel2">group 2 label</strong>
		<div>
			<span id="label3">From</span>
			<div role="group" aria-labelledby="label3">
				<input aria-label="month">/<input aria-label="day">/<input aria-label="year">
			</div>
			<span id="label4">To</span>
			<div role="group" aria-labelledby="label4">
				<input aria-label="month">/<input aria-label="day">/<input aria-label="year">
			</div>
		</div>
	</div>

</main>
</body>
</html>
```

##### EXPECTED OUTPUT:
No errors.

##### ACTUAL OUTPUT:
Multiple elements with "group" role do not have unique labels.

**Comment by tombrunet** [comment]
The main issue here is that screen readers don't necessarily tell you the hierarchy. In some situations they do, and others they don't. For example consider a layout of:

* header
  * nav
    * Link 1
* main
  * nav
    * Link 2
    
If you tab to the second link, you will hear "main region navigation region Link 2" as you expect. However, if I hit "r" through this page, it reads:
* navigation region
* main region
* navigation region

With this navigation mode, hierarchy is not conveyed. There's no difference to the user between the nested and non-nested scenario with this mode, so it doesn't make a lot of sense to have a different rule behavior for nested vs non-nested.

**Comment by wkeese** [comment]
I feared you might answer something like that, but in my test case, the "r" key in JAWS only lists the main region, which is what I'd expect because I don't think of `role=group` and `<fieldset>` as regions, or at least not landmark regions.

Are you're saying that my test case is actually problematic in a screen reader, or just that it's complicated for EAAC to figure out if it is or not?

If my test page is problematic for screen readers, do you think I should avoid using `role=group` and `<fieldset>` around related groups of fields like month/day/year?   (The brings up some philosophical questions about whether or not screenreaders are "broken", and if so, whether we should be working around that.)

**Comment by tombrunet** [comment]
I will dig a bit more into this. The next couple of weeks are a bit booked up, so will be a bit of a delay.

**Comment by tombrunet** [comment]
We've had a fair amount of debate on this. Short story is, you're right. Where I think we've settled:

In the IBM ruleset:
* group will flag as violation if a peer group/radiogroup within a form/group hierarchy has the same label.
* landmarks will flag as a violation if a peer of the same role within a landmark hierarchy has the same label.

These will flag as recommendations in the WCAG rulesets.

**Comment by aliunwala** [comment]
An update, we think this bug will get resolved by this PR: https://github.com/IBMa/equal-access/pull/541 This PR is currently working its way thought the process to get added to the Master branch. After this PR makes it, we should test your use case one more time.

**Comment by aliunwala** [comment]
This issue will hopefully be fixed once this PR goes in:
https://github.com/IBMa/equal-access/pull/642


This PR will deprecate (and replace) the rule Rpt_Aria_MultipleGroupRoles_Implicit that is firing and causing global multiple label's for groups.

**Comment by aliunwala** [comment]
And update:
Looks like these two PRs will make it to the master branch shortly, within a few days. After that I will test against the production build to see if we have solved the reported issue.

**Comment by aliunwala** [comment]
Looks like this issue made it into the master branch will be confirming this fix during the upcoming sprint.

**Comment by aliunwala** [comment]
I thought this PR would have made it to the external repo. However I did a test looks like this is working locally but not on the external chrome extension. Will have to follow up with a team member to see when the chrome extension was last updated. @eirckr 

Here is the local build without the error:
![image](https://user-images.githubusercontent.com/4451128/154399961-a2eaf614-1c98-4344-9951-8ea7db51d181.png)

And the chrome extension with the error:
![image](https://user-images.githubusercontent.com/4451128/154400113-099cd01b-3859-4329-95b1-e85c46b5d185.png)

**Comment by aliunwala** [comment]
This now works on the chrome store plugin. To replicate this you must use the preview ruleset by doing:
1) Click here:
<img width="368" alt="image" src="https://user-images.githubusercontent.com/4451128/154542847-d674edd1-ba5d-4ba0-913b-64e29d61b3a6.png">

2) Click change ruleset
<img width="642" alt="image" src="https://user-images.githubusercontent.com/4451128/154542931-4bd2828a-edbf-4757-b7e7-5317c7b5ee23.png">

3) Set the tool to preview rules:
<img width="783" alt="image" src="https://user-images.githubusercontent.com/4451128/154543015-5a5bb098-a584-4127-99c5-57c3ce16d045.png">
